### PR TITLE
mgmt azure-json, migrate azure-resourcemanager-redis

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

manual code(mainly RedisConfigurationTest migration): https://github.com/Azure/azure-sdk-for-java/commit/5e801f3973bc05f2d83eca286ec65d3ca4aa9b07

regened locally to fix `validate()`: https://github.com/Azure/azure-sdk-for-java/pull/41410/commits/deb2f5113cb7cd1521e7fc69eb842e0c56c60048

autorest PR: https://github.com/Azure/autorest.java/pull/2895

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/ConfigurationUtils.java [hunk 1]
```diff
@@ -24,6 +24,12 @@ static Map<String, String> toMap(RedisConfiguration configuration) {
             if (configuration.maxmemoryPolicy() != null) {
                 map.put("maxmemory-policy", configuration.maxmemoryPolicy());
             }
+            if (configuration.storageSubscriptionId() != null) {
+                map.put("storage-subscription-id", configuration.storageSubscriptionId());
+            }
+            if (configuration.aadEnabled() != null) {
+                map.put("aad-enabled", configuration.aadEnabled());
+            }
```
Comment (XiaofeiCao) [inline sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/implementation/ConfigurationUtils.java:32]: These changes are probably due to this test not executed during last api-version update.

**Comment by XiaofeiCao** [comment]
CI error is caused by `validate()`: https://github.com/Azure/autorest.java/issues/2890

**Review by weidongxu-microsoft — APPROVED** [review]
OK for me.

But personally on "RedisConfigurationTests.java" I would just take all class variables, remove a few by name.

Reading the java file and parse it via regex seems too fragile. It depends too much on how the code is written (not even sure if a newline would break the logic).

**Comment by XiaofeiCao** [comment]
> OK for me.
> 
> Reading the java file and parse it via regex seems too fragile. It depends too much on how the code is written (not even sure if a newline would break the logic). But I didn't see alternative though.

Agree. Both formatter and codegen logic can affect the generated code. Let's see if this is an issue in the future..

Alternative would be reading directly from swagger file based on the tag. Though it still rely on the folder/file structure.

**Pull request merged**